src/PythonPlugin.cpp

```diff
@@ -205,6 +276,82 @@ void PythonPlugin::showEditor() const
     }
 }
 
+void PythonPlugin::addScriptAction() 
+{
+    if(m_scriptList.empty())
+        m_removeScript->setEnabled(true);
+
+    QString filePath = QFileDialog::getOpenFileName(m_drawScriptRegister,
+                                              QStringLiteral("Select Python Script"),
+                                              QString(),
+                                              QStringLiteral("Python Script (*.py)"));
+    addScript(filePath);
+}
+
+void PythonPlugin::addScript(QString path) {
+    if(m_scriptList.empty())
+        m_removeScript->setEnabled(true);
+
+    QFileInfo fi(path);
+
+    //Doesn't add if file doesn't exist or if it's already present.
+    if(!fi.exists() || std::find(m_savedPath.begin(), m_savedPath.end(), path) != m_savedPath.end())
+        return;
+
+    QString fileName = fi.baseName();  
+
+    QAction* newScript = new QAction(fileName);
+    newScript->setToolTip(fileName);
+    QAction* removeNewScript = new QAction(fileName);
+    removeNewScript->setToolTip(fileName);
+
+    connect(newScript, &QAction::triggered, [this,path]()
+    {
+        executeScript(path);
+    });
+    newScript->setEnabled(true);
+    connect(removeNewScript, &QAction::triggered, [this,fileName,removeNewScript,path]()
+    {
+        removeScript(fileName,removeNewScript);
+
+        //prepare to save script list
+        int i = 0;
+        for (QString s : m_savedPath) {
+            if(s == path){
+                m_savedPath.erase(m_savedPath.begin()+i);
+                break;
+            }
+            i++;
+        }
```

Comment:
What I would do here is:

```suggestion
        auto pos = std::find(m_savedPath.begin(), m_savedPath.end(), path);
        if (pos != m_savedPath.end()){
            m_savedPath.erase(pos);
        }
```